form-validation.js

```diff
@@ -7,25 +7,29 @@ const messageMobile = document.querySelector('#messageEmail-mobile');
 const nameInput = document.querySelector('.name');
 const emailInput = document.querySelector('.email');
 const messageInput = document.querySelector('.message');
-// const formbutton = document.querySelector('#form-button');
-// const formbuttondesktop = document.querySelector('#form-button-desk');
+const rawData = localStorage.getItem('cachedFormData');
 
 const formobject = {
   name: '',
   email: '',
   message: '',
 };
 
-function storeInput() {
-  const stringobject = JSON.stringify(formobject);
-  nameInput.value = formobject.name;
+if (rawData) {
+  const formobject = JSON.parse(rawData);
   emailInput.value = formobject.email;
+  nameInput.value = formobject.name;
   messageInput.value = formobject.message;
-  localStorage.setItem('formobject', stringobject);
 }
 
-form.addEventListener('click', storeInput());
-mobileForm.addEventListener('click', storeInput());
+const dataStore = (event) => {
+  const element = event.target;
+  formobject[element.name] = element.value;
+  localStorage.setItem('cachedFormData', JSON.stringify(formobject));
+};
```

Comment:
- [ ] Great :+1: making your error message specific to the email but as requested before your error message should near the submit button without affecting its position. but as you can see from the screenshot below, while making changes your submit button has been removed from its previous place.
![image](https://user-images.githubusercontent.com/92303891/218287451-49ecd9c0-4405-403c-add2-4a38d62afdb3.png)